### DISCUSSION TITLE
Webhook

### DISCUSSION THREAD

**Discussion by Pes8** [fattureincloud/api#238] [Ideas]
Ho visto che è nella roadmap ma lo è da un bel po' di tempo ormai... si possono avere più info in merito?

- Tempistiche di rilascio
- Entità / Eventi su cui ci si potrà collegare

Ad esempio sarebbe utilissimo avere dei webhook collegati a questi eventi:

- creazione / modifica anagrafica cliente
- creazione / modifica prodotto
- creazione / modifica fattura

**Comment by emajo** [comment]
Ciao @Pes8, per quanto riguarda i webhooks siamo in una fase di testing, rilasceremo a breve l'accesso ai primi utenti interessati ma non sappiamo ancora quando verranno rilasciati pubblicamente. Gli eventi che hai citato fanno parte di quelli che andremo a notificare insieme a molti altri.

Cercheremo di avvisare nel modo più efficace gli utenti interessati e sicuramente ci faremo risentire anche qui,
Emanuele.

**Reply by Pes8** [reply]
Grazie mille, molto interessante 🙏 In caso c'è modo di essere inseriti nell'elenco degli utenti interessati per testarli?

Ancora non abbiamo sviluppato l'integrazione perchè stiamo aspettando i webhooks (per noi fondamentali per il nostro flusso) se potessimo già iniziare a testarli accorcerebbe il timing di molto

**Reply by emajo** [reply]
Certamente, appena avremo deciso il flusso per l'attivazione (credo tramite supporto clienti), avviseremo così da dare la possibilità agli interessati di avere accesso anticipato, dovrebbe trattarsi ancora di un paio di settimane di rifinitura ma non assicuro nulla perché a fine mese ci sarà la dismissione dell API v1 che ci occuperà sicuramente a tempo pieno per qualche giorno.

**Comment by Pes8** [comment]
Ciao @emajo , ci sono novità sulle tempistiche per il rilascio dei webhook ? Vorrei capire come stanno procedendo le cose

**Reply by valmoz** [reply]
Ciao @Pes8 purtroppo abbiamo avuto qualche ritardo, credo che ci vorrà ancora qualche settimana per il rilascio in alpha.

Grazie
Mauro

**Comment by danielwaghorn** [comment]
Ciao @valmoz,

We are soon to be integrating Fattureincloud with a wider system for one of our clients. If you are looking for partners to help with alpha integration and testing we would be very happy to take part.

Regards,
Dan

**Reply by valmoz** [reply]
Ok @danielwaghorn , thank you for your proposal.
we still have to decide how to open for alpha testers, so I'll keep you and @Pes8 updated on this discussion as soon as we have some additional info.

Regards,
Mauro

**Reply by valmoz** [reply]
Hi @danielwaghorn 
we sent you an email regarding your request, but we didn't receive an answer yet. Can you please check if you have received it?

Thanks
Mauro

**Comment by Pes8** [comment]
Nessuna novità ancora? (sono passati 2 mesi da quando è stato detto " _ancora qualche settimana per il rilascio in alpha_")

**Reply by valmoz** [reply]
Buongiorno,
ha ragione, purtroppo sono subentrate altre difficoltà che ci hanno costretto a rimandare il rilascio.

A brevissimo (se possibile settimana prossima, al massimo la successiva) pubblicheremo la documentazione e apriremo le richieste per accedere alla fase di testing. 

Chiediamo scusa per la lunga attesa.
Mauro

**Comment by valmoz** [comment]
🇮🇹
Buongiorno a tutti,
finalmente abbiamo appena rilasciato la documentazione relativa ai [webhooks](https://developers.fattureincloud.it/docs/webhooks/)!
Per il momento questa funzionalità sarà attiva solo per un piccolo gruppo di utenti e solo su un sottoinsieme di eventi, ma andremo progressivamente ad allargare l'elenco degli eventi pubblicati!

A partire da questo momento è possibile iniziare a richiedere l'early access ai Webhooks, compilando la Google Form che potete trovare all'interno della documentazione.

Speriamo siano di vostro gradimento! 
Il Team API di Fatture in Cloud

🇬🇧
Hello everybody,
finally, we released the [webhooks](https://developers.fattureincloud.it/docs/webhooks/) documentation!
At the moment the functionality will be available only for a small group of users and only on a small subset of events. Still, we'll progressively broaden the set of supported events!

Starting now, it will be possible to require early access to the Webhooks, by compiling the Google Form found in our documentation.

We hope you will enjoy it!
The Fatture in Cloud API Team

@Pes8 @danielwaghorn

**Reply by Pes8** [reply]
grazie per l'avviso!

**Reply by Pes8** [reply]
@valmoz occhio che il google form è chiuso: "_Questo modulo può essere visualizzato solo dagli utenti nell'organizzazione del proprietario._"

**Reply by danielwaghorn** [reply]
Grazie @valmoz! I have the same error as @Pes8; I'll register once the access is updated.

**Reply by valmoz** [reply]
Buongiorno,
scusate, adesso dovrebbe essere accessibile.

Fateci sapere
Mauro

**Reply by Pes8** [reply]
@valmoz io mi sono registrato con il form ieri pomeriggio, non mi è ancora arrivata nessuna risposta, hai idea delle tempistiche per accedere alla closed beta?

Ho provato a registrare un webhook e ovviamente ricevo il NO_PERMISSION error

**Reply by valmoz** [reply]
@Pes8 Buongiorno,
chiedo scusa, ma non avevo ancora controllato il form.
Abbiamo attivato i webhooks per la sua company.

Mauro
